base/test.jl

```diff
@@ -487,7 +491,13 @@ macro testset(args...)
     quote
         $ts = DefaultTestSet($desc)
         add_testset($ts)
-        $(esc(tests))
+        try
+            $(esc(tests))
+        catch err
+            # something in the test block threw an error. Count that as an
+            # error in this test set
+            record($ts, Error(:test_error, :(), err, catch_backtrace()))
```

Comment:
I'd call it something other than `:test_error`, so it can be distinguished from a, well, test error :D